### ISSUE TITLE
Explosion effect no longer recognized by tilesets

### ISSUE THREAD

**Issue by chaosvolt** [CleverRaven/Cataclysm-DDA#13250]
![ke bewm](https://cloud.githubusercontent.com/assets/11582235/9217239/7558db90-408a-11e5-8892-54a35fa0a618.png)

Tested in MShock and MShock Modded, unsure how someone broke explosions so hard that they're using the little round "no tile" icon, not even a fallback tile.

EDIT: Chesthole also confirmed for FUBAR, even into 3535 (0261285). I have reason to believe that SOMEONE forgot to tell us that https://github.com/CleverRaven/Cataclysm-DDA/pull/12767 broke stuff, and I can't find what I need to sprite-assign to make this headache go away.

**Comment by drbig** [comment]
@chaosvolt The explosion grading was being worked and discussed for some time already, and comparing it to #13151 is rather useless (here changes were fundamental). Having said that: yeah a way to contact all (active) tileset artists would be nice. Nobody likes fallback tiles.

**Comment by drbig** [comment]
And this isn't even a bug per se, it's strictly "missing tiles" thing (at least as I understand it).

**Comment by Chezzo** [comment]
@chaosvolt I fixed mine by adding the ids explosion_weak and explosion_medium to tile_config.

**Comment by chaosvolt** [comment]
True. I need to be more active in observing changes, but I didn't realize until viewing the PR page itself that that was the cause. Will be undoing the changes in my PR to all other tilesets except MShock Modded, and leave the other fallback tile fixing to their respective authors. Though I don't know if any of the authors besides Chezzo and I are active, so these could go unfixed for a long time. >.>

**Comment by kevingranade** [comment]
We're going to add new tile definitions from time to time, terrain, traps, items, etc.  In this case it's a "special effect" tile.  I can see how that makes it seem different, I'm sorry it caught you by surprise.

Having a fallback is a great way to have the feature never supported, and it's also rather expensive at runtime to check for fallback tiles, it's a major component of why the tile version of the game is slower than the curses version.

I'm trying to think of a way to warn about missing tiles, but the way we programattically generate them makes it difficult.  We could wrap all tile ids in a macro and use an extraction program like gettext does, I'm not sure how hard that would be to do.

**Comment by chaosvolt** [comment]
And sorry for getting all pissed about it. Most of it was being frustrated because of how long it took me to figure out what I had to add to actually fix it. I spent hours poring over the source files for animations, field effects, the explosion code etc before stumbling upon that PR that caused this. @_@

For now just adding a stopgap fix for the MShock Modded tileset, will actually take advantage of this change another day (when I set aside time for adding sprites), and will leave the responsibility of fixing the other tilesets to their respective authors.
